From b0dccaccb304541e7f56d702bfbf65e18b98c05d Mon Sep 17 00:00:00 2001
From: Ben Gamari <ben@smart-cactus.org>
Date: Tue, 10 Jan 2017 16:05:31 -0600
Subject: [PATCH] SysTools: Revert linker flags change

fefe02c0324a25a52455a61f7f6e48be6d82d1ab inadvertently removed a
linker flag needed to ensure that stack allocations don't fail
(see #8870 and #12186 for original motivation). Undo this change.

Fixes #13100.
---
 compiler/main/SysTools.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler/main/SysTools.hs b/compiler/main/SysTools.hs
index 1ab5b138e0..e5ee54d85f 100644
--- a/compiler/main/SysTools.hs
+++ b/compiler/main/SysTools.hs
@@ -860,7 +860,7 @@ getLinkerInfo' dflags = do
                      -- Note [Windows stack usage]
                      -- Force static linking of libGCC
                      -- Note [Windows static libGCC]
-                   , "-static-libgcc" ]
+                   , "-Xlinker", "--stack=0x800000,0x800000", "-static-libgcc" ]
                _ -> do
                  -- In practice, we use the compiler as the linker here. Pass
                  -- -Wl,--version to get linker version info.
-- 
2.11.0

